File: exercise/C#/day01/Communication/SantaCommunicator.cs
namespace Communication
{
    public class SantaCommunicator(int numberOfDaysToRest)
    {
        public string ComposeMessage(string reindeerName, string currentLocation, int numbersOfDaysForComingBack,
            int numberOfDaysBeforeChristmas)
        {
            var daysBeforeReturn = DaysBeforeReturn(numbersOfDaysForComingBack, numberOfDaysBeforeChristmas);
            return
                $"Dear {reindeerName}, please return from {currentLocation} in {daysBeforeReturn} day(s) to be ready and rest before Christmas.";
        }

        public bool IsOverdue(string reindeerName, string currentLocation, int numbersOfDaysForComingBack,
            int numberOfDaysBeforeChristmas, ILogger logger)
        {
            if (DaysBeforeReturn(numbersOfDaysForComingBack, numberOfDaysBeforeChristmas) <= 0)
            {
                logger.Log($"Overdue for {reindeerName} located {currentLocation}.");
                return true;
            }

            return false;
        }

        private int DaysBeforeReturn(int numbersOfDaysForComingBack, int numberOfDaysBeforeChristmas) =>
            numberOfDaysBeforeChristmas - numbersOfDaysForComingBack - numberOfDaysToRest;
    }
}

File: exercise/kotlin/day01/src/main/kotlin/communication/SantaCommunicator.kt
package communication

class SantaCommunicator(private val numberOfDaysToRest: Int) {

    fun composeMessage(
        reindeerName: String,
        currentLocation: String,
        numbersOfDaysForComingBack: Int,
        numberOfDaysBeforeChristmas: Int
    ): String {
        val daysBeforeReturn = daysBeforeReturn(numbersOfDaysForComingBack, numberOfDaysBeforeChristmas)
        return "Dear $reindeerName, please return from $currentLocation in $daysBeforeReturn day(s) to be ready and rest before Christmas."
    }

    fun isOverdue(
        reindeerName: String,
        currentLocation: String,
        numbersOfDaysForComingBack: Int,
        numberOfDaysBeforeChristmas: Int,
        logger: Logger
    ): Boolean {
        if (daysBeforeReturn(numbersOfDaysForComingBack, numberOfDaysBeforeChristmas) <= 0) {
            logger.log("Overdue for $reindeerName located $currentLocation.")
            return true
        }
        return false
    }

    private fun daysBeforeReturn(numbersOfDaysForComingBack: Int, numberOfDaysBeforeChristmas: Int): Int {
        return numberOfDaysBeforeChristmas - numbersOfDaysForComingBack - numberOfDaysToRest
    }
}